vowpalwabbit/parse_primitives.h

```diff
@@ -134,33 +115,50 @@ inline float parseFloat(char* p, char** end, char* endLine = nullptr)
   if (*p == ' ' || *p == '\n' || *p == '\t' || p == endLine)  // easy case succeeded.
   {
     acc *= powf(10, (float)(exp_acc - num_dec));
-    *end = p;
+    *end_idx = p - start;
     return s * acc;
   }
   else
-    return (float)strtod(start, end);
+  {
+    // can't use stod because that throws an exception. Use strtod instead.
+    char* end = nullptr;
+    auto ret = strtod(start, &end);
+    *end_idx = 0;
+    if (end >= start)
+    {
+      *end_idx = end - start;
+    }
+    return ret;
+  }
+    
 }
 
-inline float float_of_substring(substring s)
+inline float parse_float_string_view(VW::string_view strview, size_t& end_idx)
```

Comment:
Is it worthwhile to have this intermediate function?

Reply:
we could remove it, the syntax is just a little bit more intuitive without having to call into begin and end

Reply:
Personally, I'd remove it as a matter of style, just because being concise is good.   It's up to you.